packages/std/src/addresses.rs

```diff
@@ -58,9 +58,33 @@ impl Deref for HumanAddr {
     }
 }
 
+/// Implement `HumanAddr == str`, which gives us `&HumanAddr == &str`.
+/// Do we really need &HumanAddr comparisons?
 impl PartialEq<str> for HumanAddr {
-    fn eq(&self, other: &str) -> bool {
-        self.0 == other
+    fn eq(&self, rhs: &str) -> bool {
+        self.0 == rhs
+    }
+}
+
+/// Implement `str == HumanAddr`, which gives us `&str == &HumanAddr`.
+/// Do we really need &HumanAddr comparisons?
+impl PartialEq<HumanAddr> for str {
+    fn eq(&self, rhs: &HumanAddr) -> bool {
+        self == rhs.0
+    }
+}
+
+/// Implement `HumanAddr == &str`
+impl PartialEq<&str> for HumanAddr {
+    fn eq(&self, rhs: &&str) -> bool {
+        self.0 == *rhs
+    }
+}
+
+/// Implement `&str == HumanAddr`
+impl PartialEq<HumanAddr> for &str {
+    fn eq(&self, rhs: &HumanAddr) -> bool {
+        *self == rhs.0
```

Comment:
👆 Those two is what we really need: `HumanAddr == &str` and the other way round. When you have an `let owned_addr: HumanAddr;` you can compare it via `owned_addr == "someone"` without worrying about ownership. The `==` takes a reference from both sides and does not move the owned instance.